src/components/MapForm.js

```diff
@@ -85,15 +87,21 @@ export class MapForm extends Component {
     this.setState({ fields: { ...this.state.fields, variable } })
   }
 
-  handleTimeStart(event) {
-    const timeStart = event.target.value
-    this.setState({ fields: { ...this.state.fields, timeStart } })
+  handleChange(evt) {
+    this.setState({
+    [evt.target.name]: evt.target.value
```

Comment:
Won’t this reset all the existing state? 
I think it has to be 
`handleChange(evt) {
    This.setState({
        ...this.state,
        [evt.tatget.name]: evt.target.value
    })
}`

Reply:
not quite, here is a quick 2 min read as to why it works:
https://medium.com/better-programming/handling-multiple-form-inputs-in-react-c5eb83755d15

The images below will show state when just timeStart is adjusted, and then again after I changed timeEnd

![image](https://user-images.githubusercontent.com/13267008/68030145-c906e400-fc86-11e9-8881-ec256e710733.png)

![image](https://user-images.githubusercontent.com/13267008/68030177-dcb24a80-fc86-11e9-85a4-284a63fd2417.png)

Reply:
scratch that, our state is wrapped in 'Fields' and those images are showing NEW states, have to figure that part out